vars/mavenCanaryRelease.groovy

```diff
@@ -65,4 +71,162 @@ def call(body) {
     }
 
     contentRepository(body);
-  }
+}
+
+def patchFMPVersion() {
+    def pomModel = readMavenPom file: 'pom.xml'
+
+    if (!pomModel.profiles) {
+        return false
+    }
+
+    def updatedPlugin = false
+    for(profile in pomModel.profiles) {
+        if(!profile.id.equalsIgnoreCase("openshift") || !profile.build.plugins) {
+            continue
+        }
+
+        for(plugin in profile.build.plugins) {
+            if(plugin.artifactId != "fabric8-maven-plugin") {
+                continue
+            }
+
+            // check FMP plugin definition has '<version>' property
+            // else update FMP plugin definition <version> property according to <parent> version
+            if(plugin.version) {
+                if (compareVersions(plugin.version, FMP_STABLE_VERSION) < 0) {
+                    println "patching maven plugin for fabric8-maven-plugin v" + FMP_STABLE_VERSION
+                    plugin.version = FMP_STABLE_VERSION
+                    updatedPlugin = true
+                } else {
+                    return false
+                }
+            } else {
+                updatedPlugin = patchPluginBasedOnParent(plugin, pomModel.parent)
+            }
+        }
+    }
+
+    // if there is no FMP plugin found in pom.xml then add FMP definition based on parent module
+    if(!updatedPlugin) {
+        updatedPlugin = patchProfileBasedOnParent(pomModel)
+    }
+
+    if (updatedPlugin) {
+        writeMavenPom model: pomModel
+    }
+
+    return updatedPlugin
+}
+
+/*
+* if v1 is greater than v2 returns +ve number
+* if v2 is greater than v1 returns -ve number
+* if v1 is equal to v2 return 0
+*/
+def compareVersions(v1, v2) {
+    // not using def (v1Parts, v2Parts) = .. because UnsupportedOperationException: multiple assignments not supported
+    (v1Parts, v2Parts, maxParts) = partitionVersions(v1, v2)
```

Comment:
why return maxParts, the method guarantees that both arrays are going to be same length, doesn't it?

Reply:
How about

``` groovy
def versions = foobar()
def v1Parts = versions[0]
def v2Parts = versions[0]

```